### DISCUSSION TITLE
Unit is a lot of typing, how to define a macro in coalton that expands U into Unit

### DISCUSSION THREAD

**Discussion by devosalain** [coalton-lang/coalton#1023] [Help!] [answered]
Example : 

```

(load "~/quicklisp/setup.lisp")
(declaim (optimize (speed 3) (safety 3) (space 0) (debug 3)))
(ql:quickload "serapeum")
(ql:quickload "fiasco")
(ql:quickload "coalton")
(ql:quickload "postmodern")
(defpackage package-main
    (:use   #:cl
            #:serapeum)
    (:local-nicknames 
           (#:n #:coalton))
    (:export main)) ;defpackage
(defpackage package-coalton
    (:use   #:coalton 
            #:coalton-prelude)
    (:local-nicknames 
            (#:l #:cl) 
            (#:pm #:postmodern))
    (:export cmain) ) ;defpackage
(in-package :package-coalton)
(define-symbol-macro U Unit )                        ; Error : The variable U is unbound  !!!!!!!!!!!!!!!!!!!!!
(coalton-toplevel
    (declare myconnect ( Unit -> Unit ))
    (define (myconnect)
        (Lisp Unit () (pm:connect-toplevel "x" "x" "x" "127.0.0.1" :port 5432 ) Unit))

```

**Comment by macrologist** [comment]
`(cl:define-symbol-macro U Unit)`

**Comment by devosalain** [comment]
It does not work:

```

(defpackage package-coalton
    (:use 
        #:coalton-prelude
        #:coalton)
    (:local-nicknames 
        (#:l  #:cl)
        (#:n  #:coalton)
        (#:nl #:coalton-library/list)
        (#:ns #:coalton-library/string)
                )
    (:export cmain) );defpackage

(in-package :package-coalton)
(l:define-symbol-macro U Unit)
(coalton-toplevel
(declare getstring ( U -> String ))
    (define (getstring)
        (ns:concat astring bstring))

```

Gives error :

Unhandled L:SIMPLE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                      {1005E280A3}>:
  Unknown type U. Did you mean the type variable :U?

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1005E280A3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<L:SIMPLE-ERROR "Unknown type ~S. Did you mean the type variable ~S?" {100638F653}> #<unused argument> :QUIT L:T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<L:SIMPLE-ERROR "Unknown type ~S. Did you mean the type variable ~S?" {100638F653}>)
2: (L:INVOKE-DEBUGGER #<L:SIMPLE-ERROR "Unknown type ~S. Did you mean the type variable ~S?" {100638F653}>)
3: (L:ERROR "Unknown type ~S. Did you mean the type variable ~S?" U :U)

**Comment by stylewarning** [comment, marked as answer]
There is presently no way to make a type alias that allows one-for-one replacement.

If you wish to shorten the variable named Unit, then you can do

    (define U Unit)

but this will not work for the type Unit.

**Comment by digikar99** [comment]
> Unit is a lot of typing.

Simple autocompletion should be handled by your editor. 

That said, type aliases will certainly be useful for coalton. (Ping https://github.com/coalton-lang/coalton/issues/912.)
